cockatrice/src/gameselector.cpp

```diff
@@ -212,3 +216,17 @@ void GameSelector::processGameInfo(const ServerInfo_Game &info)
 {
     gameListModel->updateGameList(info);
 }
+
+void GameSelector::updateButtonChoices(const QModelIndex &index)
+{
+    if (!index.isValid())
+        return;
+
+    const ServerInfo_Game &game = gameListModel->getGame(index.data(Qt::UserRole).toInt());
+    bool overrideRestrictions = !tabSupervisor->getAdminLocked();
+
+    if (game.spectators_allowed() || overrideRestrictions)
```

Comment:
Simplify to `spectateButton->setEnabled(game.spectators_allowed() || overrideRestrictions)`?